### DISCUSSION TITLE
Hichiriko bamboo sax/clarinet

### DISCUSSION THREAD

**Discussion by gigaday** [edwardkort/WWIDesigner#150] [Reed Study]
First: Thanks, congratulations and respect for all the detailed and learned work that you have put into WIDesigner. Amazing!
Second: What do you make of this video of a Japanese man playing a Hichiriko bamboo sax/clarinet? https://www.youtube.com/watch?v=--JeuR1ehQQ&t=19s How does he do it? It has a great sound, good intonation, fantastic range, big evenly spaced holes. It’s a single reed instrument much like a Xaphoon. I can understand that he is using a little reverb to enhance the sound and that he is a good musician/player but there are dozens of videos of him playing and the sound is extraordinary.
Third: I used to play clarinet in a town band in Spain and inspired by the Xaphoon and other mini-saxes I thought I would have a go at making one myself. I was looking at Linsey Pollack’s book of designs and his great videos of all the instruments he has built and wondering where to start when I came across WIDesigner.
I have spent quite a few days studying the WID info, wiki, tutorials and the software – many thanks for your hard work and knowledge. And eventually set about making an eight-hole C4-C5 16mm OD x 13mm ID instrument out of PVC pipe into an alto sax mouthpiece. I did the reed calibration and tuned the bare pipe to C4. Then I did the first optimize and set all the hole sizes to be 0.5mm increments up to 8mm and optimized again – all errors 0.0. The hole spacing was too close so I set the T1-T2 hole space to 16mm to suit my fingers and then the was a small error (6 cents) each way on those two holes. Then I drilled the holes to the design as best as I could using a digital vernier, centre punch and just turning the chuck by hand – the holes are not perfect but not bad either – the smallest hole in 3.5mm and the largest 8mm.
The instrument plays OK from C4 to A4 and my phone tuner app shows it to be pretty well in tune (allowing for embouchure) and it has an acceptable (not great) sound. Any higher than that and the tuning goes off and is very embouchure dependent.
Question: What do you think I need to do to improve the sound? And get the two octave range (and more) that the Hichiriko and Xaphoon seem to be able to get? My first thought is to move up to a bigger tube diameter maybe 20mm OD or even 32mmOD.

**Comment by bhp1** [comment]
First, I'll warn you that the reed model of WIDesigner is still more or less a prototype.  It hasn't been subjected to careful trials with real instruments like you are doing.  I'd be happy to refine the model with the results of your experiments. Your mini-sax instrument sounds like a great project.

Do you have a fingering chart to work from? It looks to me that the player in the video is using some cross fingerings on the higher notes.  The xaphoon fingering chart at https://xaphoon.com/pages/fingering-charts has the lowest finger down for the first register notes from G through F.  This could be just to help stabilize the instrument, or it might make sounding the notes easier.

When you optimized the instrument, did you include the notes above C5 in the tuning file?  (Even if you didn't, though, the optimization still isn't getting B4 and C5 right.)  What fingerings would you use to get above C5 in the second register?  (Having played the clarinet, you know that overblowing will take you G5, leaving you to handle D5, E5 and F5 in the first register.)

The [note spectrum graph](https://github.com/edwardkort/WWIDesigner/wiki/Reading-the-Note-Spectrum-Graph) might give some clues why a note would be hard to sound.  For the reed study model, the graph shows only the impedance ratio, not a loop gain, but that might still offer clues where useful resonances lie.

I can't offer any suggestions on improving the tone.  A larger bore might help that, but I would expect it to make the higher notes harder to reach rather than easier.

**Comment by bhp1** [comment]
The next step in the process would be to re-calibrate the model for the instrument as-built. For each note in the scale, measure the actual frequency of the note. Put those frequencies in a tuning file (clearly labelled that it's actual frequencies, not target frequencies), and re-run the Reed Calibrator. Include second-register notes as best you can.  (If some fingerings don't sound well, you can look for alternate fingerings that do give a stable sound and include them in this tuning file, even if they are nowhere near in tune and you would never include them in your target tuning file. The extra data points help produce a calibration that works over the entire range of the instrument.)

One thing we don't yet know about the reed model: Where should we put the "mouthpiece position" and the position of the upper bore point?  Is it where the sax mouthpiece connects to the tube, the very tip of the reed in your mouth, or somewhere in between?  You might try calibrating two different instrument files that put the "mouthpiece position" around these two extremes, and see if the model does a better job of predicting the actual frequencies for one of them than the other.  I would really like to know how this trial works out.  (For the mouthpiece position at the tip of the reed, you'll have to include the inner profile of the mouthpiece in the bore profile.)

If you get a new calibration that does a decent job of predicting the actual frequencies, you can cut a new PVC tube, and use WIDesigner to lay out a new set of hole positions.

**Comment by gigaday** [comment]
Thanks for your messages. I will see what I can do to follow your instructions - it will probably be in a few days time. - Tony

**Comment by gigaday** [comment]
Ha! You make it sound so simple. I need to start a lot further back. The first thing that _I_ need to do to try and calibrate my mouth/embouchure. It’s a while since I was playing the clarinet regularly and I can get over a semi-tone variation just from varying my embouchure and changing reeds.
Before I even started seriously thinking about where to put any holes, I tuned the pipe to give me a C4 262Hz note. Then I used that for the WID reed calibrator position 0.00 “diameter” so that it was predicting the length of tube that I had already measured. I assumed that the mouthpiece was an even taper from where the tube was in the mouthpiece. I thought that 5.66mm squared was a reasonable guestimate for the tip of the reed and it did match the overall length that I was getting.
I think that I was using too aggressive an embouchure and that the instrument is now flat by about 40 cents between C4 and A5. I am wondering if I should shorten the pipe at the mouthpiece end to bring it more into tune with C4=262Hz. And see where all the other notes end up. And then go on the Mark 2 from there.
- Tony
![image](https://user-images.githubusercontent.com/1298563/172653377-2ad23d53-7a47-46e6-8848-a87d0cfa1ba0.png)
![mouthpiece and pipe](https://user-images.githubusercontent.com/1298563/172659396-9aac3097-2556-46f9-b244-11212fde63b3.png)

**Comment by bhp1** [comment]
Looking at the diagram, I would say the bore diameter is zero, not 5.66, at position zero, then increases linearly to 16.0 at somewhere short of 86.0 (maybe position 65 or so, not marked on the diagram), and remains at 16.0 to position 86.0.  At the other extreme, you could call the start of the taper (position 65 or so) as the start of the mouthpiece, and leave everything above that out of the bore model.  We don't know yet what works best for the reed model, one of these extremes or somewhere in the middle.

To calibrate an instrument, you need at least two data points: actual pitches for two notes that the instrument can produce.  The more notes, the better.  

If you've already measured frequencies for notes on the instrument you've got, I think it would be worth running the calibrator on those notes, to see how well the reed model predicts those notes, before you trim anything off the tube.  You might also try out different interpretations of the mouthpiece position to see if there's one position that works better than others.

In principle, once the reed model is good at predicting frequencies, you could use the hole spacing optimizer, _with a very carefully constructed set of constraints_, to suggest how much to trim off the tube to improve the tuning.

**Reply by gigaday** [reply]
In deciding on a point 0.0 size, I thought about the width of the reed (17mm) the tip opening (1.7mm) and then there would be the distance that the lips close the reed by (don’t know how much) – anyway I don’t think the 0.0 diameter would be zero. I know I sent you a sketch of the mouthpiece, but that was meant to be just a schematic – the actual internals of the alto sax mouthpiece are quite detailed. I assumed that the mouthpiece from the place where the tube is located is just a taper/conical. In short, I think the diameter at the tip of the mouthpiece may be just a number for any particular mouthpiece.

I did know what the overall length of the tube was (310mm) and I adjusted the value of the diameter in the WID model to be 5.66mm at 0.0 point to give me a length of 310mm when blowing a C4 note. I know this is an approximation but it turns out quite well in practice – in fact the instrument blows about a quarter tone flat throughout the first octave.

BUT – there is a huge variable that needs to be tied down before there is any point is going measuring frequencies and adjusting the WID model. And that is the embouchure! I don’t think any of the examples you have so far involve a single reed. Using varying lip pressure alone on my pipe I get pitch variations of up to 4 semitones! What I need is a standardised embouchure for myself at least.

To this end I have ordered a Maui Xaphoon in plastic and this should arrive later today. If I can get it to play in tune, I will have something to calibrate my embouchure to and then I can start thinking about modelling similar instruments.

You may be interested to see how Brian Wittman makes a Xaphoon out of bamboo. He uses a red-hot poker (literally) to make the holes and he says he knows from experience where to put them. https://www.youtube.com/watch?v=jbzrTcZ37aI and then he adjusts the hole to complete the tuning. He makes it look very easy. The Maui Xaphoon is his plastic version of the instrument, which he says sounds very much like the bamboo version. I am assuming that he went though an optimisation process before he committed to a production run of his Maui Xaphoon.

Tony

**Comment by bhp1** [comment]
It sounds to me that you have used the diameter of the top bore point to "calibrate" the model so it predicted your C4 note.  The intention is that the diameter of the top bore point represents the real, measured, geometry.  We calibrate the model by adjusting the alpha and beta factors, which are _not_ measurable attributes of the geometry.  
![image](https://user-images.githubusercontent.com/1645842/173621534-9db727bc-eeec-4b16-9127-edc935770604.png)
You can adjust alpha and beta manually, or if you have measured frequencies for two or more notes, you can use the Reed Calibrator.  (If you don't have at least two notes, you can't determine a unique pair of values of alpha and beta that calibrate the model.)

Your point about calibrating your embouchure is a good one.  The model will be aiming its predictions at the same embouchure and blowing pattern that you use for your calibration measurements, so you'll want to be conscious of how you're blowing, and make an effort to be consistent.

**Reply by gigaday** [reply]
I am lost with the alpha and beta factors. The Glossary does make me much wiser. Where should I be looking for more info?

If I manually change the values of alpha and beta it makes no difference to the 310mm length value that I have determined by trial using my electronic tuner.

**Reply by bhp1** [reply]
Alpha and beta are abstract tuning factors of the model, not directly connected to something you can measure in the real world.  They adjust for all the elements of mouthpiece tuning that we can't capture just by measuring the geometry:  things like the reed profile and reed stiffness, even how hard your embouchure is and how hard you tend to blow.

With beta set to zero, try setting alpha to 1, then 2, then 3, and calculate the tuning table each time.  When I do this, I see the predicted frequency of C4 going about 50 cents flatter with each increase in alpha (and higher notes get flatter by much greater amounts).  With alpha set to 1.0, try setting beta to 0.0, then 0.1, then 0.2.  When I do this, I see the predicted frequency of C4 (and other notes, in general) going about 20 cents flatter with each increase in beta.

The "correct" values of alpha and beta are those that make the predicted values as close as possible to measured values on the real instrument.

**Comment by bhp1** [comment]
A minor point: Never use the same bore point position for two different bore points, to avoid confusing them.  I suggest you change the two bore points at 86.0 to have one at 86.0 and one at 86.1:

![image](https://user-images.githubusercontent.com/1645842/173874315-3f6436d3-4057-4530-bea9-1fd9d6b4c213.png)

**Reply by gigaday** [reply]
Thanks for your input on these points. I am not going to do anything further on my 16mm pipe model for now. I am considering my options for building another pipe, possible with greater wall thickness - but that may not be for a while. I will need go through all the steps again and I don't suppose that I will remember everything.

I received the Maui Xaphoon the day before yesterday. It is certainly quite hard to play, which I think may be due to the wide hole spacing (42mm) between T1 and T2. But it does seem to play more in tune than other reed pipes that I have tried. The embouchure is tough and I will need to persevere with it until I get some of my lip strength back after not playing reeds for a few years.

He has used a nine hole design, which makes sense from the point of view of using all available fingers but also makes the fingering less like say a recorder as the Xaphoon uses the left hand little finger for G.
